core/src/test/java/com/datastax/oss/driver/internal/core/metadata/DefaultTopologyMonitorTest.java

```diff
@@ -329,7 +328,121 @@ public void should_refresh_node_list_from_local_and_peers() {
   }
 
   @Test
-  public void should_stop_executing_queries_once_closed() throws Exception {
+  public void should_refresh_node_list_from_local_and_peers_v2() {
+    // Given
+    AdminRow peer3 = mockPeersV2Row(3, UUID.randomUUID());
+    AdminRow peer2 = mockPeersV2Row(2, node2.getHostId());
+    topologyMonitor.stubQueries(
+        new StubbedQuery("SELECT * FROM system.local", mockResult(mockLocalRow(1))),
+        new StubbedQuery("SELECT * FROM system.peers_v2", mockResult(peer3, peer2)));
+
+    // When
+    CompletionStage<Iterable<NodeInfo>> futureInfos = topologyMonitor.refreshNodeList();
+
+    // Then
+    assertThatStage(futureInfos)
+        .isSuccess(
+            infos -> {
+              Iterator<NodeInfo> iterator = infos.iterator();
+              NodeInfo info1 = iterator.next();
+              assertThat(info1.getEndPoint()).isEqualTo(node1.getEndPoint());
+              assertThat(info1.getDatacenter()).isEqualTo("dc1");
+              NodeInfo info3 = iterator.next();
+              assertThat(info3.getEndPoint().resolve())
+                  .isEqualTo(new InetSocketAddress("127.0.0.3", 9042));
+              assertThat(info3.getDatacenter()).isEqualTo("dc3");
+              NodeInfo info2 = iterator.next();
+              assertThat(info2.getEndPoint()).isEqualTo(node2.getEndPoint());
+              assertThat(info2.getDatacenter()).isEqualTo("dc2");
+            });
+  }
+
+  @Test
+  public void should_skip_invalid_peers_row() {
+    // Given
+    topologyMonitor.isSchemaV2 = false;
+    node2.broadcastAddress = ADDRESS2;
+    AdminRow peer2 = mockPeersRow(2, node2.getHostId());
+    when(peer2.isNotEmpty("rpc_address")).thenReturn(false);
+    topologyMonitor.stubQueries(
+        new StubbedQuery(
+            "SELECT * FROM system.peers WHERE peer = :address",
+            ImmutableMap.of("address", ADDRESS2.getAddress()),
+            mockResult(peer2)));
+
+    // When
+    CompletionStage<Optional<NodeInfo>> futureInfo = topologyMonitor.refreshNode(node2);
+
+    // Then
+    assertThatStage(futureInfo).isSuccess(maybeInfo -> assertThat(maybeInfo).isEmpty());
+    assertThat(node2.broadcastAddress).isNotNull().isEqualTo(ADDRESS2);
+  }
+
+  @Test
+  public void should_skip_invalid_peers_row_with_extended_check() {
+    // Given
+    topologyMonitor.isSchemaV2 = false;
+    node2.broadcastAddress = ADDRESS2;
+    AdminRow peer2 = mockPeersRow(2, node2.getHostId());
+    when(peer2.isNotEmpty("tokens")).thenReturn(false);
```

Comment:
shouldn't we create a parameterized test for all columns that we checking with `isNotEmpty`?

Reply:
Good idea, will do.